### PR TITLE
Added Retry Context for python worker

### PR DESCRIPTION
**Pull request by gavin-aguiar**
## Description
- Implemented retry context of python worker. 
Doc: https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-error-pages?tabs=python#retry-policies-preview

- Update protobuf file to include latest changed from https://github.com/Azure/azure-functions-language-worker-protobuf/blob/dev/src/proto/FunctionRpc.proto

Fixes #

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azure_functions_worker/dispatcher.py [hunk 1]
```diff
@@ -346,13 +344,11 @@ async def _handle__function_load_request(self, req):
...
         invoc_request = req.invocation_request
-
         invocation_id = invoc_request.invocation_id
         function_id = invoc_request.function_id
-        trace_context = bindings.TraceContext(
-            invoc_request.trace_context.trace_parent,
-            invoc_request.trace_context.trace_state,
-            invoc_request.trace_context.attributes)
+        trace_context, retry_context = \
+            self._get_trace_and_retry_context(invoc_request)
```
Comment (vrdmr) [inline azure_functions_worker/dispatcher.py:350]: You can name it as `get_context` as in the future there might be more contexts might come and we don't want to change this again.

File: azure_functions_worker/dispatcher.py [hunk 2]
```diff
@@ -556,6 +553,19 @@ async def _handle__close_shared_memory_resources_request(self, req):
                 request_id=self.request_id,
                 close_shared_memory_resources_response=response)
 
+    @staticmethod
+    def _get_trace_and_retry_context(invocation_request):
```
Comment (vrdmr) [inline azure_functions_worker/dispatcher.py:557]: Is there any check needed if any one of the contexts is None? Why set an object as None itself?

Comment (vrdmr) [reply azure_functions_worker/dispatcher.py:557]: On second thought - can the whole Context object be made here? Bring in this code, within this function?
```
bindings.Context(
                fi.name, fi.directory, invocation_id,
                trace_context, retry_context)
```

You already have all the contexts. Also, get the context closer to where it's used - near L391?

**Review by vrdmr — APPROVED** [review]
LGTM
🚢  😄

**Comment by odaom** [comment]
Has this worked its way into a non-preview release yet?

**Pull request merged**